### ISSUE TITLE
[FEATURE] infer typings from parameters

### ISSUE THREAD

**Issue by p-mcgowan** [acrontum/generate-it#41]
**Is your feature request related to a problem? Please describe.**
When defining post body params, typings are `any`

for instance, the resulting yaml:
```yaml
paths:
  /somethings:
    get:
      parameters:
        - $ref: '#/parameters/a'
        - $ref: '#/parameters/b'
        - $ref: '#/parameters/c'
      responses:
        '200':
          description: Ok
          schema:
            $ref: '#/definitions/SomethingModels'
        '404':
          description: Not found
    post:
      operationId: somethingPost
      parameters:
        - in: body
          name: something
          required: true
          schema:
            type: object
            required:
              - code
              - name
            properties:
              code:
                type: string
              name:
                type: string
              appId:
                type: number
```

the get route produces the correct typings, but the post route (since it has no definition models) is not inferred from the spec.
in this case, it seems like it should be able to produce the typings, possibly as follows:
```typescript
export interface SomethingsPostBody {
  code: string;
  name: string;
  appId?: number;
}
```

**Describe the solution you'd like**
Depending on the extent of the solution, the ideal situation would be a path/method/paramtype named interface with the properties defined

**Describe alternatives you've considered**
The alternative is to make sure each body has a separate definition - this would be ok, but could require defining different types for post, patch, update. without template extension, this would likely mean much duplication, and maintenance.

**Comment by Celleb** [comment]
As it currently works, anything that needs a type in generate-it must be put in definitions.

This would work:

```yaml
parameters:
  - in: body
    name: somethingPostBody
    description: The fields to create the user with
    schema:
      $ref: ../../definitions/user/post.yml
```

**Comment by p-mcgowan** [comment]
Yeah, that was one of the alternatives - however, in this case, if you have GET POST PATCH, all of which have only some fields required (maybe different ones for each route), you end up with the base model, plus 3 more for params.

It would be really cool if the typings could be inferred from the provided schema - then arbitrary one-offs would still have typings in the domains.
